### PR TITLE
Cleanup minor issues in S.S.C.Encryption left from initial port

### LINKED ISSUE #1753: Style changes for System.Security.Cryptography.Encryption
Some style-only comments were given in pull request https://github.com/dotnet/corefx/pull/1746.

To best facilitate the code round-tripping through the mirror this issue was created to do the cleanup after merging to master:

- [x] Reduce the configuration matrix down to just Debug and Release, since it has no OS-specific code. (https://github.com/dotnet/corefx/pull/1746#discussion_r30343867).
- [x] CryptoStream has some redundant initializers, which should be removed. (https://github.com/dotnet/corefx/pull/1746#discussion_r30344072)
- [x] CryptoStream has some private fields which are _PascalCased instead of _camelCased. (https://github.com/dotnet/corefx/pull/1746#discussion_r30344205)
- [x] CryptoStream has a trivial overload, why is it there? (https://github.com/dotnet/corefx/pull/1746#discussion_r30344517) (maybe just that "adding an overload later is a recompile-required change")

### PR DESCRIPTION
**Pull request by stephentoub**
Variables named with wrong casing, unnecessary overrides that just delegate to base, unnecessary usings, and platform-specific configurations for a platform-agnostic assembly.

Fixes #1753

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @bartonjs

File: src/System.Security.Cryptography.Encryption/src/System/Security/Cryptography/CryptoStream.cs [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
 using System.Runtime.InteropServices;
-using System.Diagnostics.Contracts;
-using System.Threading;
-using System.Threading.Tasks;
+
 
 namespace System.Security.Cryptography
 {
-    [System.Runtime.InteropServices.ComVisible(true)]
+    [ComVisible(true)]
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Encryption/src/System/Security/Cryptography/CryptoStream.cs]: Is ComVisible even applicable in CoreFX?

**Comment by bartonjs** [comment]
LGTM (assuming, of course, that CI agrees)

File: src/System.Security.Cryptography.Encryption/src/System/Security/Cryptography/CryptoStream.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Security.Cryptography.Encryption/src/System/Security/Cryptography/CryptoStream.cs]: Nope.  I previously scrubbed out a bunch of these.  I'll make another pass to do so again, as some more have snuck in (like this one).

**Pull request merged**